# Dataset Understanding

## Project
Predictive Maintenance System for Manufacturing Equipment

### Objective

The objective of this notebook is to understand the structure, characteristics, and engineering significance of the AI4I 2020 Predictive Maintenance dataset before performing exploratory analysis and machine learning modeling.

### Tasks

1. Load dataset
2. Inspect dataset structure
3. Study variables
4. Analyze data quality
5. Understand engineering significance of each feature

In [ ]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)
from google.colab import files

uploaded = files.upload()

Saving ai4i2020.csv to ai4i2020.csv


In [ ]:
df = pd.read_csv("ai4i2020.csv")
df.head()

,UDI,Product ID,Type,Air temperature [K],Process temperature [K],Rotational speed [rpm],Torque [Nm],Tool wear [min],Machine failure,TWF,HDF,PWF,OSF,RNF
0,1,M14860,M,298.1,308.6,1551,42.8,0,0,0,0,0,0,0
1,2,L47181,L,298.2,308.7,1408,46.3,3,0,0,0,0,0,0
2,3,L47182,L,298.1,308.5,1498,49.4,5,0,0,0,0,0,0
3,4,L47183,L,298.2,308.6,1433,39.5,7,0,0,0,0,0,0
4,5,L47184,L,298.2,308.7,1408,40.0,9,0,0,0,0,0,0


In [ ]:
df.info()
df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   UDI                      10000 non-null  int64  
 1   Product ID               10000 non-null  object 
 2   Type                     10000 non-null  object 
 3   Air temperature [K]      10000 non-null  float64
 4   Process temperature [K]  10000 non-null  float64
 5   Rotational speed [rpm]   10000 non-null  int64  
 6   Torque [Nm]              10000 non-null  float64
 7   Tool wear [min]          10000 non-null  int64  
 8   Machine failure          10000 non-null  int64  
 9   TWF                      10000 non-null  int64  
 10  HDF                      10000 non-null  int64  
 11  PWF                      10000 non-null  int64  
 12  OSF                      10000 non-null  int64  
 13  RNF                      10000 non-null  int64  
dtypes: float64(3), int64(9)

,0
UDI,0
Product ID,0
Type,0
Air temperature [K],0
Process temperature [K],0
Rotational speed [rpm],0
Torque [Nm],0
Tool wear [min],0
Machine failure,0
TWF,0


## **Dataset Structure Analysis**

The dataset contains 10,000 observations and 14 variables. The variables consist of machine identification information, operational sensor measurements, and failure indicators.

A total of 12 variables are numerical while 2 variables are categorical. The dataset contains no missing values, indicating high data quality and eliminating the need for imputation techniques during preprocessing.

The target variable, **Machine failure**, is binary, where a value of 1 represents machine failure and 0 represents normal operation.

The dataset is therefore suitable for supervised machine learning classification tasks aimed at predicting equipment failures.


## **Missing Value Analysis**

An inspection of missing values revealed that all variables contain complete observations. No missing values were detected in the dataset.

This is advantageous because model development can proceed without applying missing-value treatment methods such as deletion or imputation.

The absence of missing values also suggests that the dataset has been well curated for predictive maintenance research and educational applications.




In [ ]:
df.describe()

,UDI,Air temperature [K],Process temperature [K],Rotational speed [rpm],Torque [Nm],Tool wear [min],Machine failure,TWF,HDF,PWF,OSF,RNF
count,10000.00000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.00000
mean,5000.50000,300.004930,310.005560,1538.776100,39.986910,107.951000,0.033900,0.004600,0.011500,0.009500,0.009800,0.00190
std,2886.89568,2.000259,1.483734,179.284096,9.968934,63.654147,0.180981,0.067671,0.106625,0.097009,0.098514,0.04355
min,1.00000,295.300000,305.700000,1168.000000,3.800000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.00000
25%,2500.75000,298.300000,308.800000,1423.000000,33.200000,53.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.00000
50%,5000.50000,300.100000,310.100000,1503.000000,40.100000,108.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.00000
75%,7500.25000,301.500000,311.100000,1612.000000,46.800000,162.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.00000
max,10000.00000,304.500000,313.800000,2886.000000,76.600000,253.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.00000


## Statistical Observations

The rotational speed ranges between **1168** and **2886** RPM.

Torque ranges between **3.8** and **76.6** Nm.

Tool wear ranges between **0** and **253** minutes.

## Engineering Interpretation of Variables

### UDI

Unique identifier assigned to each observation. This variable has no engineering significance and will not contribute to machine failure prediction.

### Product ID

Unique identifier representing a specific machine product. Primarily used for tracking and identification purposes.

### Type

Machine quality category:

* L = Low Quality
* M = Medium Quality
* H = High Quality

Machine type may influence reliability, operational behavior, and failure patterns.

### Air Temperature [K]

Represents the ambient environmental temperature surrounding the machine.

Elevated ambient temperatures may increase thermal stress and influence machine operating conditions.

### Process Temperature [K]

Represents the temperature of the manufacturing process.

Abnormally high process temperatures may indicate excessive friction, inefficient operation, or developing mechanical issues.

### Rotational Speed [rpm]

Represents shaft rotational speed measured in revolutions per minute.

Rotational speed influences vibration characteristics, bearing loads, fatigue behavior, and mechanical wear.

### Torque [Nm]

Represents mechanical loading applied to the machine.

Higher torque values generally correspond to increased stress on machine components and may accelerate wear mechanisms.

### Tool Wear [min]

Represents accumulated tool usage time.

Greater tool wear indicates progressive degradation and potentially higher failure risk.

### Machine Failure

Target variable indicating whether machine failure occurred.

Value 0 = Normal operation

Value 1 = Failure

### Failure Modes

The following variables identify specific failure mechanisms:

* TWF = Tool Wear Failure
* HDF = Heat Dissipation Failure
* PWF = Power Failure
* OSF = Overstrain Failure
* RNF = Random Failure

These variables provide additional insight into the underlying causes of machine failure.


In [20]:
df["Machine failure"].value_counts(normalize=True)*100

,proportion
Machine failure,
0,96.61
1,3.39


## **Target Variable Analysis**

The target variable, Machine Failure, exhibits significant class imbalance.

Out of 10,000 observations, **9,661** instances correspond to normal machine operation, while only **339** instances represent machine failures.

This results in a failure rate of approximately **3.39%**.

Such imbalance is commonly observed in industrial predictive maintenance applications because equipment failures are relatively rare compared to normal operating conditions.

The class imbalance indicates that model evaluation should not rely solely on accuracy. Additional metrics such as precision, recall, F1-score, and confusion matrices will be necessary to assess predictive performance effectively.
